src/software/thunderscope/robot_diagnostics/robot_view.py

```diff
@@ -17,168 +15,44 @@ class RobotView(QWidget):
 
     """
 
-    toggle_robot_connection_signal = QtCore.pyqtSignal(int)
+    toggle_all_connection_signal = QtCore.pyqtSignal(int, int)
```

Comment:
Im finding this naming a bit confusing. It seems more like robot connection changed signal than _toggle_ for _all_ robots

Reply:
done, i'm calling it control_mode_signal now since it's not just about if robots are connected, but rather which control mode they are currently set to